plugins/auth-backend-module-bitbucket-server-provider/src/authenticator.ts

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2024 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Strategy as OAuth2Strategy, VerifyCallback } from 'passport-oauth2';
+import {
+  createOAuthAuthenticator,
+  PassportOAuthAuthenticatorHelper,
+  PassportProfile,
+} from '@backstage/plugin-auth-node';
+
+/** @public */
+export const bitbucketServerAuthenticator = createOAuthAuthenticator({
+  defaultProfileTransform:
```

Comment:
It looks like the profile was fetched in a specific way before, should it be replicated here in this transform? The legacy fetch profile function had access to the config, how can I access it here?

Reply:
I updated the logic to fetch the profile manually, please take a look.